Add list groups for akhq

nerhneiro opened the pull request:
### Changelog entry <!-- a user-readable short description of the changes that goes to CHANGELOG.md and Release Notes -->

KafkaListGroupsActor for AKHQ

### Changelog category <!-- remove all except one -->

* Not for changelog (changelog entry is not required)

### Description for reviewers <!-- (optional) description for those who read this PR -->

...

Commit:
generate TListGroupRequestData and TListGroupResponseData classes, EApiKey::LIST_GROUPS

Commit:
add actor files, generate TListGroupsResponseData and TListGroupsRequestData in kaffka_messages.h
write test to check is ListGroupsActor works correctly

Commit:
add statesFilter realization
add new test

a-serebryanskiy commented on ydb/core/kafka_proxy/kafka_connection.cpp:
```diff
@@ -312,6 +312,10 @@ class TKafkaConnection: public TActorBootstrapped<TKafkaConnection>, public TNet
     }
 
+    void HandleMessage(const TRequestHeaderData* header, const TMessagePtr<TListGroupsRequestData>& message) {
+        Register(CreateKafkaListGroupsActor(Context, header->CorrelationId, message));
```
Тут надо проверять наличие фича флага, включающего использование таблиц для балансирвоки консьюмеров (EnableKafkaNativeBalancing) и если его нет, то возвращать ошибку UNSUPPORTED_VERSION и писать в лог сообщение, поясняющее ошибку. 

Получить фича флаг можно вот так: https://github.com/ydb-platform/ydb/blob/main/ydb/core/kafka_proxy/kafka_connection.cpp#L243

a-serebryanskiy commented on ydb/core/kafka_proxy/actors/kafka_list_groups_actor.cpp:
```diff
@@ -0,0 +1,221 @@
+
+void TKafkaListGroupsActor::Handle(NKqp::TEvKqp::TEvQueryResponse::TPtr& ev, const TActorContext& ctx) {
+    KAFKA_LOG_D("Received query response from KQP ListGroups request");
+    if (auto error = GetErrorFromYdbResponse(ev)) {
```
Может возникнуть ситуация, когда запрос на лист групп пришел, а табличка еще не создана, потому что табличка создается при первом запросе в kafka_balance_actor. 
Нужно:
а) понять, какая ошибка от KQP придет, если таблицы не существует и в случае такой ошибки отправить запрос на создание таблицы, как это делается вот тут https://github.com/ydb-platform/ydb/blob/main/ydb/core/kafka_proxy/actors/kafka_balancer_actor.cpp#L25
б) добавить в наш с тобой e2e тест запрос консьюмер групп до создания первой консьюмер группы в тесте. После этого запроса нужна проверка, что вернулся пустой массив.

a-serebryanskiy commented on ydb/core/kafka_proxy/ut/ut_protocol.cpp:
```diff
@@ -1857,6 +1857,123 @@ Y_UNIT_TEST_SUITE(KafkaProtocol) {
+            UNIT_ASSERT_VALUES_EQUAL(*group.GroupState, "CompletingRebalance");
+            UNIT_ASSERT_VALUES_EQUAL(*group.ProtocolType, protocolType);
+
+            std::cout << "********" << std::endl;
```
Обычно используем вместе std::cout просто Cout и Endl

a-serebryanskiy commented on ydb/core/kafka_proxy/ut/ut_protocol.cpp:
```diff
@@ -1857,6 +1857,123 @@ Y_UNIT_TEST_SUITE(KafkaProtocol) {
+
+        TListGroupsRequestData requestGroupsStateFilter;
+        requestGroupsStateFilter.StatesFilter.push_back("PreparingRebalance");
+        // requestGroupsStateFilter.StatesFilter.push_back("CompletingRebalance");
```
протек комментарий

a-serebryanskiy commented on ydb/core/kafka_proxy/ut/ut_protocol.cpp:
```diff
@@ -1857,6 +1857,123 @@ Y_UNIT_TEST_SUITE(KafkaProtocol) {
+                UNIT_ASSERT_VALUES_EQUAL(*group.GroupState, "CompletingRebalance");
+            } else if (*group.GroupId == groupId2) {
+                check_both_groups_exist *= 3;
+                UNIT_ASSERT_VALUES_EQUAL(*group.GroupState, "PreparingRebalance"); // check that Group2 is in "preparing rebalance" state
```
коммент тут кажется лишний, все понятно из кода

a-serebryanskiy commented on ydb/core/kafka_proxy/ut/ut_protocol.cpp:
```diff
@@ -1857,6 +1857,123 @@ Y_UNIT_TEST_SUITE(KafkaProtocol) {
+        }
+        UNIT_ASSERT_VALUES_EQUAL(check_both_groups_exist, 6);
+
+        TListGroupsRequestData requestGroupsStateFilter;
```
А вот тут бы я добавил коммент о том, какую логику мы планируем проверить дальше

Commit:
add correct DatabasePath from context
add check on EnableKafkaNativeBalancing feature flag existance
add empty check in up_protocol.cpp
correct codestyle
delete excesive variables
add comments

Commit:
add ApiKey for LIST_GROUPS
change error check from NOT_FOUND to SCHEME_ERROR

Commit:
Merge branch 'main' into add-list-groups-for-akhq

Commit:
Merge branch 'main' into add-list-groups-for-akhq

Commit:
change apiKeys size in checks, add correct Die for kafka_list_groups_actor

Commit:
Merge branch 'main' into add-list-groups-for-akhq

Commit:
add correct EXPECTED_API_KEYS_COUNT
